### PR TITLE
Use `esy command-env` to get the Esy target directory

### PR DESCRIPTION
**Pull request by anmonteiro**
This is better than a hardcoded path because it works in all esy
versions. Furthermore, it also makes RLS work with Esy aliased sandboxes

### CODE REVIEW

File: util/Files.re [hunk 1]
```diff
@@ -157,7 +157,9 @@ let exists = (path) =>
   | Some(_) => true
   };
 
-let ifExists = path => exists(path) ? Some(path) : None;
+let ifExists = fun
```
Comment (jaredly) [inline util/Files.re]: hmmm I'd rather not change the signature of this function

File: src/analyze/BuildSystem.re [hunk 2]
```diff
@@ -124,27 +124,52 @@ let detect = (rootPath, bsconfig) => {
...
+        }
+      | err => err
+      }
     }
-  | _ => assert(false)
+  | _ => Error("Couldn't find Esy target directory")
   }
 
 let getCompiledBase = (root, buildSystem) =>
   Files.ifExists(
```
Comment (jaredly) [inline src/analyze/BuildSystem.re]: how bout you do the switch outside of this

File: src/analyze/BuildSystem.re [hunk 3]
```diff
@@ -124,27 +124,52 @@ let detect = (rootPath, bsconfig) => {
...
+let getEsyCompiledBase = (root, esyVersion) =>
   switch (String.split_on_char('.', esyVersion)) {
   | [_, minor, ..._] =>
-    if(int_of_string(minor) >= 3) {
-      "_esy" /+ "default" /+ "build"
-    } else {
-      "_build" /* TODO */
+    let env = Unix.environment () |. Array.to_list;
+    try {
+      let root = List.find(var => Utils.startsWith(var, "cur__original_root="), env);
```
Comment (jaredly) [inline src/analyze/BuildSystem.re]: maybe ditch this part and go right to the `execResult`? This function should only be run on startup

Comment (anmonteiro) [reply src/analyze/BuildSystem.re]: We wanna check the environment to support the aliased sandboxes use case. Why do you suggest skipping this check?

Comment (jaredly) [reply src/analyze/BuildSystem.re]: I guess I'm confused about why vscode's environment would have these variables. Do you expect people to start vscode with `esy code .` or similar?

Comment (jaredly) [reply src/analyze/BuildSystem.re]: huhh ok!

**Comment by anmonteiro** [comment]
@jaredly I addressed the outstanding code review items. LMK if there's anything else.

**Pull request merged**